### PR TITLE
Fix: Bug in MRMFeatureQCFile

### PR DESCRIPTION
**Pull request by dmccloskey**
# Description
- Small bug in MRMFeatureQCFile where an empty vector in a member of MRMFeatureQC would throw an error

### CODE REVIEW

**Comment by dmccloskey** [comment]
I just realized that this PR is pulling in two commits from a recent PR that was merged.  Let me know if this is a problem for the merging strategy or review and I can cherry pick in a new PR.

**Comment by timosachsenberg** [comment]
no problem. I can try an automated rebase.
/rebase

File: src/openms/source/FORMAT/MRMFeatureQCFile.cpp [hunk 1]
```diff
@@ -211,6 +211,8 @@ namespace OpenMS
   void MRMFeatureQCFile::store(const String & filename, const MRMFeatureQC & mrmfqc, const bool is_component_group)
   {
     if (is_component_group) {
+      if (mrmfqc.component_group_qcs.size() <= 0) return;
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/MRMFeatureQCFile.cpp:214]: .empty() is probably better here and below

**Review by timosachsenberg — APPROVED** [review]
If you could add a regression test as well it would be great.

File: src/openms/source/FORMAT/MRMFeatureQCFile.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/MRMFeatureQCFile.cpp:214]: I will add those changes

**Pull request merged**